File: Lab9-23/test/total/total.py
i0, j0, l0 = map(int, input().split())
check = False

for k in range(0, 51):

    if i0 <= 0 and j0 >= -10:
        if j0 <= 20 and j0 >= 0:
            if j0 > 10:
                if j0 <= -i0 + 10:
                    K = k
                    check = True
                    break
            else:
                if j0 >= i0 + 10:
                    K = k
                    check = True
                    break
    
    i = min(max(min(i0 - j0, i0 - l0), j0 - l0), i0 - k) % 30
    j = max(min(max(i0 - j0, i0 - l0), j0 - l0), i0 - k) % 30
    l = i0 % 30 - j0 % 30 + l0 % 30 - k % 30

    i0, j0, l0 = i, j, l

    print(f'k = {k}, i = {i0}, j = {j0}, l = {l0}')


if check:
    print(f'Hit. k = {K}, i = {i0}, j = {j0}, l = {l0}')
else:
    print(f'Miss. i = {i0}, j = {j0}, l = {l0}')

File: Lab9-23/test/total/total.c
#include <stdio.h>
#include <stdbool.h>
#include "../../mathlib.h"

int main() {

    bool check = false;
    int i0, j0, l0;
    scanf("%d %d %d", &i0, &j0, &l0);
    int i, j, l;
    int K;

    for (int k = 0; k <= 50; k++) {
        if (in_triangle(i0, j0)) {
            check = true;
            K = k;
            break;
        }

        i = mod(min(max(min(i0 - j0, i0 - l0), j0 - l0), i0 - k), 30);
        j = mod(max(min(max(i0 - j0, i0 - l0), j0 - l0), i0 - k), 30);
        l = mod(i0, 30) - mod(j0, 30) + mod(l0, 30) - mod(k, 30);

        i0 = i;
        j0 = j;
        l0 = l;

        printf("k = %d, i = %d, j = %d, l = %d\n", k, i, j ,l);
    }

    if (check)
        printf("Hit. k = %d, i = %d, j = %d, l = %d\n", K, i0, j0, l0);
    else
        printf("Miss. i = %d, j = %d, l = %d\n", i0, j0, l0);
}